helix-core/src/main/java/org/apache/helix/controller/stages/TopStateHandoffReportStage.java

```diff
@@ -491,6 +491,12 @@ private void reportTopStateComesBack(ResourceControllerDataProvider cache, Map<S
                 true);
       }
     }
+
+    // In case of recovery after failure, we should decrement the missingTopStateBeyondThresholdGauge value.
+    if (clusterStatusMonitor != null && record.isFailed()) {
```

Comment:
First and foremost, congratulations on your first change.
Q: why do we decrement only for recovery after failure scenario and not in-general recovery?
Q: why don't i see change for 1min related reset window?

Reply:
Had offline conversation. A: to first one is that whenever we cross threshold, we mark it FAILED.
A: to the second one is:this is generic reset - not specific to the metric in question.

Reply:
Also to second one, we want to keep fallback backup reset mechanism to reset the metric.